lib/RVSDG/RVSDGOps.cpp

```diff
@@ -455,52 +502,62 @@ LogicalResult PhiResult::verify() {
+        << " Number of region arguments: " << regionArgTypes.size();
   }
   size_t typeIndex = 0;
-  for (auto [inType, argType] :zip(inputTypes, regionArgTypes)) {
-    if (inType != argType) {
+  for (auto [inType, argType] : zip(inputTypes, regionArgTypes))
+  {
+    if (inType != argType)
+    {
       return emitOpError("Type mismatch between node inputs and region arguments.")
-            << " Offending argument: #"
-            << typeIndex 
-            << " Input type: " << inType
-            << " Region argument type: " << argType;
+          << " Offending argument: #" << typeIndex << " Input type: " << inType
+          << " Region argument type: " << argType;
     }
     ++typeIndex;
   }
   return LogicalResult::success();
 }
 
 /*
-* Delta result verifier.
-* Verifies the following attributes:
-* - Number and types of operands match the output of the parent delta node.
-*   - Delta output is a typed pointer which references the value given to the operand,
-*     so the operand type should match the element type of the output.
-*/
-LogicalResult DeltaResult::verify() {
+ * Delta result verifier.
+ * Verifies the following attributes:
+ * - Number and types of operands match the output of the parent delta node.
+ *   - Delta output is a typed pointer which references the value given to the operand,
+ *     so the operand type should match the element type of the output.
+ */
+LogicalResult
+DeltaResult::verify()
+{
   auto parent = dyn_cast<DeltaNode>((*this)->getParentOp());
-  if (parent == NULL) {
+  if (parent == NULL)
+  {
     return emitOpError("DeltaResult has no parent of type DeltaNode.");
   }
-  auto resultType = this->getOperand().getType();
-  auto outputType = parent.getOutput().getType();
-  mlir::Type outputElementType;
-  if (auto rvsdgPtrType = outputType.dyn_cast_or_null<RVSDGPointerType>()) {
-    outputElementType = rvsdgPtrType.getElementType();
-  }
-  if (resultType != outputElementType) {
-    return emitOpError("Type mismatch between DeltaResult and DeltaNode output.")
-           << " DeltaResult type: " << resultType
-           << " DeltaNode output element type: " << outputElementType;
-  }
+  // auto resultType = this->getOperand().getType();
+  // auto outputType = parent.getOutput().getType();
+  // mlir::Type outputElementType;
+  // if (auto rvsdgPtrType = outputType.dyn_cast_or_null<RVSDGPointerType>()) {
+  //   outputElementType = rvsdgPtrType.getElementType();
+  // }
+  // All pointers are opaque in LLVM 18 so this check does not make sense
+  //  else if (auto llvmPtrType = outputType.dyn_cast_or_null<mlir::LLVM::LLVMPointerType>()) {
+  //    outputElementType = llvmPtrType.getElementType();
+  //  }
+  // if (resultType != outputElementType) {
+  //   return emitOpError("Type mismatch between DeltaResult and DeltaNode output.")
+  //          << " DeltaResult type: " << resultType
+  //          << " DeltaNode output element type: " << outputElementType;
+  // }
```

Comment:
Why is validation of DeltaResults commented out?

Reply:
Delta outputs an opaque pointer, nothing to validate